### PR TITLE
First Draft

### PR DESCRIPTION
**Pull request by scottcreynolds**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1 +1,78 @@
-const app = "I don't do much."
+var cake = {
```
Comment (pletcher) [inline index.js:1]: I like how this file sets up students with code to debug -- great practice!

File: test/index-test.js [hunk 2]
```diff
@@ -6,13 +6,152 @@ const path = require('path')
...
+      var node = document.getElementById("cake")
+      makeCake.call(node)
+      expect(spy).toHaveBeenCalledWith(node)
+      spy.restore()
+    })
+    it('calls mix with context', () => {
+      var spy = expect.spyOn(mix, "call").andCallThrough()
+      var node = document.getElementById("cake")
+      makeCake.call(node)
+      expect(spy.calls[0].arguments[0]).toBe(cake)
```
Comment (pletcher) [inline test/index-test.js]: Would `expect(spy).toHaveBeenCalledWith(cake)` work?

File: README.md [hunk 3]
```diff
@@ -9,43 +9,61 @@ JavaScript Fix this
...
+* Create a version of `updateStatus` inside of `makePie` and `makeCake` with a correct `this` context, representing either the `pie` or `cake` `<div>` respectively, that you can pass around to the other functions so that each one can execute it and ensure that the right DOM elements are getting updated at each step. You shouldn't need to change `updateStatus` at all.
 
-Another potential test manipulates things in the DOM, but messes up the `this` in the callback
+* There's also some bugs in the code, so make sure the tests are passing and run the page to see it in action and make sure it works! Some tests already pass. Part of your job is to make sure they still pass at the end!
+
+Each step of the way, you should be using what you know about `call`,
+`bind`, and `apply` to make each function set the context for the next
+function.
+
+**Advanced:** If you get tired of manually binding context to every
```
Comment (pletcher) [inline README.md]: As with `solution`, we can just kind of mention this :)

**Comment by pletcher** [comment]
This looks great! Just a couple of comments :)

File: test/index-test.js (hunk 2 above)
Comment (scottcreynolds) [inline test/index-test.js]: `toHaveBeenCalledWith` needs all args and I was only testing for the one, but easy enough to bind an update function and check against it, probably a better test.

File: index.js (hunk 1 above)
Comment (scottcreynolds) [inline index.js:1]: Thanks!. Can never have enough practice getting into existing code to fix bugs or add features.

**Comment by scottcreynolds** [comment]
@pletcher okay updated here and solution

**Comment by pletcher** [comment]
Thanks! ⚓ s aweigh!

**Pull request merged**